.circleci/config.yml

```diff
@@ -2,94 +2,183 @@
     steps:
-      - checkout
-      - restore_cache: { key: 'gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
-      - restore_cache: { key: 'gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
-      - run: ./gradlew --no-daemon --parallel --continue build
+      - attach_workspace: { at: /home/circleci }
+      - restore_cache: { key: 'gradle-wrapper-v2-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
+      - restore_cache: { key: 'check-gradle-cache-v2-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - run: ./gradlew --parallel --stacktrace --continue check idea -x test
+      - save_cache:
+          key: 'check-gradle-cache-v2-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}'
+          paths: [ ~/.gradle/caches ]
       - run:
           command: mkdir -p ~/junit && find . -type f -regex ".*/build/.*TEST.*xml" -exec cp --parents {} ~/junit/ \;
           when: always
       - store_test_results: { path: ~/junit }
-      - store_artifacts: { path: ~/poms }
+      - store_artifacts: { path: ~/artifacts }
 
-  build-ibm:
-    docker: [{ image: 'ellisjoe/ibm-java-openssl:java-8' }]
-    resource_class: xlarge
+  unit-test:
+    docker: [{ image: 'cimg/openjdk:11.0.10-node' }]
+    resource_class: large
     environment:
-      GRADLE_OPTS: -Dorg.gradle.console=plain -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false
-
+      CIRCLE_TEST_REPORTS: /home/circleci/junit
+      CIRCLE_ARTIFACTS: /home/circleci/artifacts
+      GRADLE_OPTS: -Dorg.gradle.jvmargs='-Xmx2g' -Dorg.gradle.workers.max=2
+      _JAVA_OPTIONS: -XX:ActiveProcessorCount=4 -XX:MaxRAM=8g -XX:ErrorFile=/home/circleci/artifacts/hs_err_pid%p.log -XX:HeapDumpPath=/home/circleci/artifacts
     steps:
-      - checkout
-      - restore_cache: { key: 'gradle-wrapper-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
-      - restore_cache: { key: 'gradle-cache-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
-      - run: ./gradlew --no-daemon --parallel --continue crypto-core:test
+      - attach_workspace: { at: /home/circleci }
+      - restore_cache: { key: 'gradle-wrapper-v2-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
+      - restore_cache: { key: 'unit-test-gradle-cache-v2-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - run:
+          name: link_openssl
+          command: |
+            sudo apt-get update -q && sudo apt-get install -qy libssl-dev
+            sudo ln -s /lib/x86_64-linux-gnu/libcrypt.so /usr/lib/libcrypto.so
+            sudo ln -s /lib/x86_64-linux-gnu/libcrypt.so /usr/lib64/libcrypto.so
+            echo "Linked openssl /usr/lib/libcrypto.so"
+            ls -al /lib/libcrypto* /usr/lib/libcrypto* /usr/lib64/libcrypto* /usr/lib/x86_64-linux-gnu/libcrypto*
```

Comment:
This is the section we'll need to figure out how we want to fold into circle template so that it doesn't immediately get removed by excavator though unit tests should block that from merging, but that also means the excavator won't be updating everything else including CI images.

For future reference, see https://issues.apache.org/jira/browse/HADOOP-12845 & https://issues.apache.org/jira/browse/HADOOP-11216

Resulting libcrypto symlinks (we need `/usr/lib64/libcrypto.so`)
```
lrwxrwxrwx 1 root root      33 Jan  6 20:36 /lib/libcrypto.so -> /lib/x86_64-linux-gnu/libcrypt.so
lrwxrwxrwx 1 root root      33 Jan  6 20:36 /usr/lib64/libcrypto.so -> /lib/x86_64-linux-gnu/libcrypt.so
lrwxrwxrwx 1 root root      33 Jan  6 20:36 /usr/lib/libcrypto.so -> /lib/x86_64-linux-gnu/libcrypt.so
-rw-r--r-- 1 root root 5541876 Nov 24 13:20 /usr/lib/x86_64-linux-gnu/libcrypto.a
lrwxrwxrwx 1 root root      16 Nov 24 13:20 /usr/lib/x86_64-linux-gnu/libcrypto.so -> libcrypto.so.1.1
-rw-r--r-- 1 root root 2954080 Nov 24 13:20 /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
```